misk/src/main/kotlin/misk/CoordinatedService.kt

```diff
@@ -118,11 +118,46 @@ internal class CoordinatedService(val service: Service) : AbstractService() {
         }
       }
 
+      val validityMap = mutableMapOf<CoordinatedService, CycleValidity>()
+      for (service in coordinatedServices) {
+        val cycle = service.findCycle(validityMap)
+        if (cycle != null) {
+          errors.add("dependency cycle: ${cycle.joinToString(separator = " -> ")}")
```

Comment:
Maybe return some kind of Error class or even a string so that it's not the client's responsibility to traverse the cycle? Or so you see a use for the cycle list?

Reply:
In this case this is a programmer-facing error message. You should be able to tell that your EggService can’t depend on your ChickenService.